### PR TITLE
[proj4] Update to 8.1.1, revise features and dependencies

### LINKED ISSUE #19910: [proj4] update to 8.1.1
Library name: proj4

New version number: 8.1.1

Other information that may be useful (release notes, etc...)
https://github.com/OSGeo/PROJ/releases/latest

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?  
  - Update proj to 8.1.1. (Resolves #19910.)
  - Adds a new `net` feature. It is enabled by default, following upstream.
    Network support (via libcurl) was bound to the `tools` feature but in fact it is independent.
    (I wonder if there is an established name for such a feature. Alternatives are "network" or "online".)
  - Enables tiff and network support for static builds. This is essential for the official osx and linux triplets.
    The key issue was properly exporting the dependencies for consumers (cmake config and pkgconfig).

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  unchanged

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

#### Important behavioural change

The opt-in feature `tools` no longer implies network support.

### CODE REVIEW

**Comment by dg0yt** [comment]
Okay, libspatialite needs to learn transitive dependencies.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by dg0yt** [comment]
Depends on #20480.

File: ports/proj4/portfile.cmake [hunk 1]
```diff
@@ -12,41 +12,37 @@ vcpkg_from_github(
...
-  set(TOOL_NAMES cct cs2cs geod gie proj projinfo projsync)
+vcpkg_list(SET TOOL_NAMES cct cs2cs geod gie proj projinfo)
+if("net" IN_LIST FEATURES)
+    vcpkg_list(APPEND TOOL_NAMES projsync)
+    set(feature_for_projsync "tools")
 else()
-  set(VCPKG_BUILD_SHARED_LIBS OFF)
-  set(TOOL_NAMES cct cs2cs geod gie proj projinfo)
+    set(feature_for_projsync "net")
 endif()
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/proj4/portfile.cmake]: Rewritten.

**Comment by dg0yt** [comment]
Seems to need #21594 which in turn needs #21599 which needs review from @JackBoosY.

**Comment by dg0yt** [comment]
libcrypto from openssl needs user32.lib. This is not an issue with CMake which has implicit link libraries. But it is an isse with nmake when relying only on pkg-config.
~~~
libcrypto.lib(cryptlib.obj) : error LNK2019: unresolved external symbol __imp_GetProcessWindowStation referenced in function OPENSSL_isservice
libcrypto.lib(cryptlib.obj) : error LNK2019: unresolved external symbol __imp_GetUserObjectInformationW referenced in function OPENSSL_isservice
libcrypto.lib(cryptlib.obj) : error LNK2019: unresolved external symbol __imp_MessageBoxW referenced in function OPENSSL_showfatal
spatialite.dll : fatal error LNK1120: 3 unresolved externals
~~~

Alternative A: Update libcrypto.pc. Rebuilds all dependencies of openssl. 
Alternative B: Pass cmake's implicit link libraries to nmake. Rebuilds all dependencies of vcpkg_build_nmake.

I tend towards B. It seems to more consistent overall behaviour in vcpkg.

**Comment by Neumann-A** [comment]
@dg0yt But them into `ENV{_LINK_}` just like `vcpkg_configure_make`?

**Comment by dg0yt** [comment]
> @dg0yt But them into `ENV{_LINK_}` just like `vcpkg_configure_make`? (or `ENV{LIBS}` if nmake uses that...)

Something like `vcpkg_configure_make` does, for B, yes.

**Comment by dg0yt** [comment]
> Alternative A: Update libcrypto.pc. Rebuilds all dependencies of openssl. 
> Alternative B: Pass cmake's implicit link libraries to nmake. Rebuilds all dependencies of vcpkg_build_nmake.

To unblock the highly desired proj changes *now*, I implemented alternative C: Add user32.lib explicitly where needed.

*In the long run*, I still prefer alternative B. But this would be a fundamental change. It would better go to a new port vcpkg-nmake instead of the currrent unversioned maintainer script.

**Comment by dg0yt** [comment]
CI strangeness: On osx, it rebuilds some ports which should be cached and are not affected by the changes. From the logs:

[Build 1](https://dev.azure.com/vcpkg/public/_build/results?buildId=64268&view=results), 0f07f0f: Rebuilding readosm, 
~~~
Agent: vcpkg-eg-mac-03

HEAD is now at 65433e8c3 Merge 0f07f0fe0c231c9e8da14def0f28642caf46283b into 431d3810aa9241aed49b14367be417f225d2af5e


                          freexl:x64-osx:     pass: 3336081133f426c8aa88c625fd0efa8c6f03cd9d481226770fb0c0abfa94bb5f
                         readosm:x64-osx:     pass: bd8d2ef7219f20a0174d386abcd73fc717429c326efe2be276a9e7c83f75416b

2021-12-07T12:15:13.8180670Z Uploaded binaries to 1 HTTP remotes.
2021-12-07T12:15:13.8183800Z Installing package readosm[core]:x64-osx...
2021-12-07T12:15:13.8256250Z Elapsed time for package readosm:x64-osx: 31.13 s
~~~

[Build 2](https://dev.azure.com/vcpkg/public/_build/results?buildId=64269&view=results), d5bc878: Rebuiding freexl
~~~
Agent: vcpkg-eg-mac-03

HEAD is now at b5978228c Merge d5bc878bd0c63f843139520a133519e3f0b0963a into 431d3810aa9241aed49b14367be417f225d2af5e

                          freexl:x64-osx:     pass: 3336081133f426c8aa88c625fd0efa8c6f03cd9d481226770fb0c0abfa94bb5f
                         readosm:x64-osx:     pass: bd8d2ef7219f20a0174d386abcd73fc717429c326efe2be276a9e7c83f75416b

2021-12-07T12:25:59.1985960Z Uploaded binaries to 1 HTTP remotes.
2021-12-07T12:25:59.1988430Z Installing package freexl[core]:x64-osx...
2021-12-07T12:25:59.2056200Z Elapsed time for package freexl:x64-osx: 41.47 s
~~~

Build 2 was started after completing build 1, the ABI hashes didn't change, and `master` didn't  change. Neither freexl nor readosm depend on the changed ports.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the updates!

**Pull request merged**